examples/src/lib.rs

```diff
@@ -173,32 +180,39 @@ pub struct ExampleBase {
 
 impl ExampleBase {
     pub fn render_loop<F: Fn()>(&self, f: F) {
-        use winit::*;
-        self.events_loop.borrow_mut().run_forever(|event| {
-            f();
-            match event {
-                Event::WindowEvent { event, .. } => match event {
-                    WindowEvent::KeyboardInput { input, .. } => {
-                        if let Some(VirtualKeyCode::Escape) = input.virtual_keycode {
-                            ControlFlow::Break
-                        } else {
-                            ControlFlow::Continue
-                        }
-                    }
-                    WindowEvent::CloseRequested => winit::ControlFlow::Break,
-                    _ => ControlFlow::Continue,
-                },
-                _ => ControlFlow::Continue,
-            }
-        });
+        self.events_loop
+            .borrow_mut()
+            .run_return(|event, _, control_flow| {
+                *control_flow = ControlFlow::Wait;
+                f();
+                match event {
+                    Event::WindowEvent {
+                        ref event,
+                        window_id,
+                    } if window_id == self.window.id() => match event {
```

Comment:
Is `if window_id == self.window.id()` actually necessary here? This isn't a multi-window application.

Reply:
It's good practice to showcase it in an example, though.

Reply:
I don't think I agree; including it suggests to a naive reader that it does something meaningful in this case, which is misleading, and multi-window applications are rare regardless.

Reply:
@Ralith What do you think? Shall we just remove this `if` and get on with it? One downside of this PR is that it uses `EventLoopExtRunReturn` which isn't supported on at least IOS (maybe also MacOS?) but at least it is _way_ less intrusive than all the other PRs that refactor the entire example while at it (necessary evil to have a place _inside_ the eventloop to free all resources :disappointed:).

Will need to do some title/commitbody curating though before this is mergeable.

Reply:
I think we should merge this, and I'd prefer that we do so with the guard removed (but it's a big improvement regardless).

Reply:
Fair enough. Fortunately @neurotok left this branch push-able for maintainers so I've just removed that, recombined the nested `matches` into one big one, and turned it into an `if-let` as per clippy's request. Should be ready to merge then, and we can worry about the other PRs when someone really wants to run this example on iOS. I think that's a necessary tradeoff to make for the gross of issues out there running older winit.

Reply:
Yeah, I'm a big fan of incremental refactoring in general--helps manage reviewer bandwidth if nothing else :D

Reply:
Definitely, small isolated PRs are _a lot_ easier to process and the few extra minutes taken to create multiple PRs (not mentioning the act of separating ones changes during development, that's a whole different rabbit hole!) or clicking multiple merge buttons completely outweigh the _hours_ otherwise spent comprehending massive changesets :D